src/main/java/frc/robot/subsystems/BuddyBalanceSubsystem.java

```diff
@@ -0,0 +1,88 @@
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.hardware.MotorController;
+import frc.robot.hardware.MotorController.MotorConfig;
+import frc.robot.classes.TunableNumber;
+
+import javax.management.relation.Relation;
+
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.RelativeEncoder;
+import com.revrobotics.SparkMaxPIDController;
+
+public class BuddyBalanceSubsystem extends SubsystemBase { // TODO: Find a permanent name for buddy balance subsystem - BBalance is ambiguous
+  public static final int buddyBalanceRefPointDocked = 50;
+  public static final int buddyBalanceRefPointBalanced = 15; // TODO: Replace constants with actual ref points during testing
+  public static final int buddyBalanceRefPointDeployed = 0;
+
+  private CANSparkMax rightMotor;
+  private CANSparkMax leftMotor;
+  private SparkMaxPIDController rightPIDController;
+  private SparkMaxPIDController leftPIDController;
+  private RelativeEncoder rightEncoder;
+  private RelativeEncoder leftEncoder;
+  private DataLog datalog = DataLogManager.getLog();
+  private DoubleLogEntry buddyBalancePosition = new DoubleLogEntry(datalog, "/buddybalance/position");
+  private ShuffleboardTab buddyBalanceTab = Shuffleboard.getTab("Buddy Balance"); // TODO: Replace buddy balance tab with whatever tab the position should be logged to
+  private GenericEntry positionEntry;
+  private boolean isToggled = false;
+  private TunableNumber tNumBuddyBalanceRightP;
+  private TunableNumber tNumBuddyBalanceRightI;
+  private TunableNumber tNumBuddyBalanceRightD;
+  private TunableNumber tNumBuddyBalanceLeftP;
+  private TunableNumber tNumBuddyBalanceLeftI;
+  private TunableNumber tNumBuddyBalanceLeftD;
+
+  public BuddyBalanceSubsystem() {
+    rightMotor = MotorController.constructMotor(MotorConfig.BBalanceRightShaftRotate);
+    leftMotor = MotorController.constructMotor(MotorConfig.BBalanceLeftShaftRotate);
+    rightPIDController = rightMotor.getPIDController();
+    leftPIDController = leftMotor.getPIDController();
+    rightEncoder = rightMotor.getEncoder();
+    leftEncoder = leftMotor.getEncoder();
+    rightPIDController.setReference(0, CANSparkMax.ControlType.kPosition);
+    leftPIDController.setReference(0, CANSparkMax.ControlType.kPosition);
+
+    tNumBuddyBalanceRightP = new TunableNumber("Buddy Balance Right Motor P", 0, rightPIDController::setP);
+    tNumBuddyBalanceRightI = new TunableNumber("Buddy Balance Right Motor I", 0, rightPIDController::setI);
+    tNumBuddyBalanceRightD = new TunableNumber("Buddy Balance Right Motor D", 0, rightPIDController::setD);
+    tNumBuddyBalanceLeftP = new TunableNumber("Buddy Balance Left Motor P", 0, leftPIDController::setP);
+    tNumBuddyBalanceLeftI = new TunableNumber("Buddy Balance Left Motor I", 0, leftPIDController::setI);
+    tNumBuddyBalanceLeftD = new TunableNumber("Buddy Balance Left Motor D", 0, leftPIDController::setD);
```

Comment:
Set the P constants to something >0. I'd also recommend making D something non-zero as well.